ska_conda/ska_builder.py

```diff
@@ -11,32 +11,56 @@ class SkaBuilder(object):
 
     def __init__(self, ska_root=None):
         if ska_root is None:
-            ska_root = "/data/acis/ska3_pkg/"
+            ska_root = "/tmp/ska3_pkg/"
         self.ska_root = ska_root
         self.ska_build_dir = os.path.join(self.ska_root, "builds")
         self.ska_src_dir = os.path.join(self.ska_root, "src")
         os.environ["SKA_TOP_SRC_DIR"] = self.ska_src_dir
 
-    def _clone_repo(self, name):
+    def _clone_repo(self, name, tag=None):
         if name == "ska":
             return
         print("Cloning source %s." % name)
         clone_path = os.path.join(self.ska_src_dir, name)
-        if os.path.exists(clone_path):
-            print("Source %s exists, skipping." % name)
-            return
-        yml = os.path.join(pkg_defs_path, name, "meta.yaml")
-        with open(yml) as f:
-            requires = False
-            while not requires:
-                line = f.readline().strip()
-                if line.startswith("path:"):
-                    requires = True
-            data = yaml.load(f)
-            url = data['about']['home']
-            git.Repo.clone_from(url, clone_path)
+        if not os.path.exists(clone_path):
+            # Try ssh first to avoid needing passwords for the private repos
+            # We could add these ssh strings to the meta.yaml for convenience
+            try:
+                git_ssh_path = 'git@github.com:sot/' + name + '.git'
+                repo = git.Repo.clone_from(git_ssh_path, clone_path)
+                assert not repo.bare
+            except:
+                yml = os.path.join(pkg_defs_path, name, "meta.yaml")
+                with open(yml) as f:
+                    requires = False
+                    while not requires:
+                        line = f.readline().strip()
+                        if line.startswith("path:"):
+                            requires = True
+                    data = yaml.load(f)
+                    url = data['about']['home']
+                repo = git.Repo.clone_from(url, clone_path)
+        else:
+            repo = git.Repo(clone_path)
+            repo.remotes.origin.fetch()
+            repo.remotes.origin.fetch("--tags")
```

Comment:
As far as I could tell, the fetching behavior seems to differ a bit based on git version, and I'm not sure about gitpython at all.  Using both fetch and fetch --tags seemed, at worse, duplication.
I think that fetching from 'origin' will be appropriate in all cases.

Reply:
We can use conda git for this right?  That should make behavior uniform.  Git fetch origin alone will always get the tags.

Reply:
I think gitpython gets the git in your path.  And yes, conda git should probably work as the thing it gets in the path.  Should get added to requirements (and I don't recall which environments/installs had issues with https).     Just when trying to figure out how you do a fetch with gitpython, I had also seen the language that "most" tags should be reachable via `git fetch` and hadn't figured the exceptions yet.

Reply:
It looks to me that we need "--tags" to be safe if a tag has actually changed on origin.  For example, I retagged skare3 repo and I wasn't getting the associated new commit until adding this back again.  I know this shouldn't come up (you should re-use a tag), but...

Could also delete all local tags before the fetch, but that seems more problematic.